src/OpenTelemetry.AutoInstrumentation/Logging/OtelLogging.cs

```diff
@@ -0,0 +1,133 @@
+﻿// <copyright file="OtelLogging.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using OpenTelemetry.AutoInstrumentation.Configuration;
+
+namespace OpenTelemetry.AutoInstrumentation.Logging;
+
+/// <summary>
+/// Configures shared logger used by instrumentations.
+/// </summary>
+internal static class OtelLogging
+{
+    private const string NixDefaultDirectory = "/var/log/opentelemetry/dotnet";
+
+    private static readonly ILogger Logger;
+
+    static OtelLogging()
+    {
+        ISink sink = new NoopSink();
+        try
+        {
+            var logDirectory = GetLogDirectory();
+            if (logDirectory != null)
+            {
+                var fileName = GetLogFileName();
+                var logPath = Path.Combine(logDirectory, fileName);
+                sink = new FileSink(logPath);
+            }
+        }
+        catch (Exception)
+        {
+            // ignored
+        }
+
+        Logger = new Logger(sink);
+    }
+
+    internal static ILogger GetLogger() => Logger;
+
+    private static string GetLogFileName()
+    {
+        try
+        {
+            using var process = Process.GetCurrentProcess();
+            return $"dotnet-tracer-managed-{process.ProcessName}-{process.Id}.log";
+        }
+        catch
+        {
+            // We can't get the process info
+            return "dotnet-tracer-managed.log";
```

Comment:
you could use a timestamp instead of guid but but then outside of this function you would need to handle the case if the file is already owned by other process.